### PR TITLE
Add Parameter Server Example to CI Tests

### PR DESCRIPTION
**Pull request by pbelevich**
Resolves one item in #46321

Run test:
pytest test/distributed/rpc/test_tensorpipe_agent.py -k test_parameter_server -vs
pytest test/distributed/rpc/test_process_group_agent.py -k test_parameter_server -vs

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51037 Add RPC Pipeline Example to CI Tests
* #51016 Add RPC RNN Example to CI Tests
* **#50930 Add Parameter Server Example to CI Tests**

Differential Revision: [D26049857](https://our.internmc.facebook.com/intern/diff/D26049857)

### CODE REVIEW

File: torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py [hunk 1]
```diff
@@ -1,139 +1,252 @@
...
+
+from time import perf_counter
+from threading import Lock
 
-from torch.testing._internal.dist_utils import (
-    dist_init,
-    worker_name,
-)
+from torch.testing._internal.dist_utils import dist_init
 from torch.testing._internal.distributed.rpc.rpc_agent_test_fixture import RpcAgentTestFixture
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py:17]: shall we sort the import block

File: torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py [hunk 2]
```diff
@@ -1,139 +1,252 @@
...
+
+# Given an RRef, return the result of calling the passed in method on the value
+# held by the RRef. This call is done on the remote node that owns
+# the RRef. args and kwargs are passed into the method.
+# Example: If the value held by the RRef is of type Foo, then
+# remote_method(Foo.bar, rref, arg1, arg2) is equivalent to calling
+# <foo_instance>.bar(arg1, arg2) on the remote node and getting the result
+# back.
+
+
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py:79]: unintentional new lines?

File: torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py [hunk 3]
```diff
@@ -1,139 +1,252 @@
...
+            # Ensure that dist autograd ran successfully and gradients were
+            # returned.
+            assert remote_method(
+                ParameterServer.get_dist_gradients,
+                net.param_server_rref,
+                cid) != {}
+            opt.step(cid)
+
+    print("Training complete!")
+    print("Getting accuracy....")
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py:183]: does it make sense to suppress some outputs as this is not user facing.

File: torch/testing/_internal/distributed/rpc/rpc_agent_test_fixture.py [hunk 4]
```diff
@@ -5,6 +5,10 @@
 
 
 class RpcAgentTestFixture(ABC):
+    @property
+    def name(self):
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/rpc/rpc_agent_test_fixture.py:9]: we can use the existing API:

https://github.com/pytorch/pytorch/blob/f0e72e54cc3049e986adc5bff314dd0b649c780b/torch/testing/_internal/dist_utils.py#L194

File: torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py [hunk 5]
```diff
@@ -1,139 +1,252 @@
...
+    param_rrefs = net.get_global_param_rrefs()
+    opt = DistributedOptimizer(optim.SGD, param_rrefs, lr=0.03)
+    for i, (data, target) in enumerate(train_loader):
+        with dist_autograd.context() as cid:
+            model_output = net(data)
+            target = target.to(model_output.device)
+            loss = F.nll_loss(model_output, target)
+            if i % 5 == 0:
+                print(f"Rank {rank} training batch {i} loss {loss.item()}")
+            dist_autograd.backward(cid, [loss])
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py:173]: is it possible to fix input and model init values (say by fixing rand seed), and then have some assert on the loss? (e.g., after a few iterations, the loss should drop)

**Review by mrshenli — APPROVED** [review]
LGTM! Added some minor comments.

File: torch/testing/_internal/distributed/rpc/rpc_agent_test_fixture.py (hunk 4 above)
Comment (pbelevich) [reply torch/testing/_internal/distributed/rpc/rpc_agent_test_fixture.py:9]: addressed

File: torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py (hunk 3 above)
Comment (pbelevich) [reply torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py:183]: But what if someone needs to debug this test, will one need to add it back?

File: torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py (hunk 2 above)
Comment (pbelevich) [reply torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py:79]: removed, literally copy-pasted from the original source

File: torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py (hunk 1 above)
Comment (pbelevich) [reply torch/testing/_internal/distributed/rpc/examples/parameter_server_test.py:17]: addressed
